Riot/Modules/Settings/Security/SecurityViewController.m

```diff
@@ -290,6 +297,105 @@ - (void)viewWillDisappear:(BOOL)animated
+    NSMutableArray<Section*> *sections = [NSMutableArray array];
+    
+    // Pin code section
+    
+    Section *pinCodeSection = [Section sectionWithTag:SECTION_PIN_CODE];
+    
+    // Header title
+    NSString *pinCodeSectionHeaderTitleFormat = NSLocalizedStringFromTable(@"pin_protection_settings_section_header_x", @"Vector", nil);
+    NSString *pinCodeSectionHeaderTitle = [NSString stringWithFormat:pinCodeSectionHeaderTitleFormat, [PinCodePreferences shared].localizedBiometricsName];
+    pinCodeSection.headerTitle = pinCodeSectionHeaderTitle;
+    
+    // Rows
+    [pinCodeSection addRowWithTag:PIN_CODE_SETTING];
+    [pinCodeSection addRowWithTag:PIN_CODE_DESCRIPTION];
+    
+    if (![PinCodePreferences shared].isBiometricsAvailable)
+    {
+        [pinCodeSection addRowWithTag:PIN_CODE_BIOMETRICS];
+    }
+    
+    [sections addObject:pinCodeSection];
+    
+    // Crypto sessions section
+        
+    Section *sessionsSection = [Section sectionWithTag:SECTION_CRYPTO_SESSIONS];
+    
+    sessionsSection.headerTitle = NSLocalizedStringFromTable(@"security_settings_crypto_sessions", @"Vector", nil);
+        
+    NSUInteger sessionsSectionRowsCount;
+    
+    if (self.showLoadingDevicesInformation)
+    {
+        sessionsSectionRowsCount = 2;
+    } else {
+        sessionsSectionRowsCount = devicesArray.count + 1;
+    }
+
+    [sessionsSection addRowsWithCount:sessionsSectionRowsCount];
+    
+    [sections addObject:sessionsSection];
+    
+    // Secure backup
+    
+    
+    Section *secureBackupSection = [Section sectionWithTag:SECTION_SECURE_BACKUP];
+    secureBackupSection.headerTitle = NSLocalizedStringFromTable(@"security_settings_secure_backup", @"Vector", nil);
+    
+    [secureBackupSection addRowsWithCount:[self numberOfRowsInSecureBackupSection]];
+    
+    [sections addObject:secureBackupSection];
+    
+    // Cryptograhpy
+    
+    Section *cryptograhpySection = [Section sectionWithTag:SECTION_CRYPTOGRAPHY];
+    cryptograhpySection.headerTitle = NSLocalizedStringFromTable(@"security_settings_cryptography", @"Vector", nil);
+    
+    [cryptograhpySection addRowsWithCount:CRYPTOGRAPHY_COUNT];
+    
+    [sections addObject:cryptograhpySection];
+    
+#ifdef CROSS_SIGNING_AND_BACKUP_DEV
+    
+    // Cross-Signing
+    
+    Section *crossSigningSection = [Section sectionWithTag:SECTION_CROSSSIGNING];
+    crossSigningSection.headerTitle = NSLocalizedStringFromTable(@"security_settings_crosssigning", @"Vector", nil);
+    
+    [crossSigningSection addRowsWithCount:[self numberOfRowsInCrossSigningSection]];
+    
+    [sections addObject:crossSigningSection];
+    
+    // Keybackup
+    
+    Section *keybackupSection = [Section sectionWithTag:SECTION_KEYBACKUP];
+    keybackupSection.headerTitle = NSLocalizedStringFromTable(@"security_settings_backup", @"Vector", nil);
+    
+    [keybackupSection addRowsWithCount:[self keyBackupSection.numberOfRows]];
+    
+    [sections addObject:keybackupSection];
```

Comment:
I think we should call `updateSections` instead of `tableView.reloadData` in `settingsKeyBackupTableViewSectionDidUpdate` callback.